File: software/bindings/java/ExampleInterrupt.java
import com.tinkerforge.BrickletIO4;
import com.tinkerforge.IPConnection;

public class ExampleInterrupt {
	private static final String host = new String("localhost");
	private static final int port = 4223;
	private static final String UID = new String("ABC"); // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(host, port); // Create connection to brickd (Can throw IOException)

		BrickletIO4 io4 = new BrickletIO4(UID); // Create device object
		ipcon.addDevice(io4); // Add device to ip connection (Can throw IPConnection.TimeoutException)
		// Don't use device before it is added to a connection
		

		// Enable interrupt on pin 0
		io4.setInterrupt((short)(1 << 0));

		// Add and implement listener for interrupt (called if pin 0 changes)
		io4.addListener(new BrickletIO4.InterruptListener() {
			public void interrupt(short interruptMask, short valueMask) {
				System.out.println("Interrupt by: " + Integer.toBinaryString(interruptMask));
				System.out.println("Value: " + Integer.toBinaryString(valueMask));
			}
		});
		
		System.out.println("Press ctrl+c to exit");
		ipcon.joinThread();
	}
}

File: software/bindings/csharp/ExampleInterrupt.cs
using Tinkerforge;

class Example
{
	private static string HOST = "localhost";
	private static int PORT = 4223;
	private static string UID = "ABC"; // Change to your UID

	// Callback function for interrupts
	static void InterruptCB(byte interruptMask, byte valueMask)
	{
		string interruptBinary = System.Convert.ToString(interruptMask, 2);
		string valueBinary = System.Convert.ToString(valueMask, 2);

		System.Console.WriteLine("Interrupt by: " + interruptBinary);
		System.Console.WriteLine("Value: " + valueBinary);
	}

	static void Main() 
	{
		IPConnection ipcon = new IPConnection(HOST, PORT); // Create connection to brickd
		BrickletIO4 io4 = new BrickletIO4(UID); // Create device object
		ipcon.AddDevice(io4); // Add device to ip connection
		// Don't use device before it is added to a connection

		// Enable interrupt on pin 0
		io4.SetInterrupt(1 << 0);

		// Register callback for interrupts
		io4.RegisterCallback(new BrickletIO4.Interrupt(InterruptCB));

		System.Console.WriteLine("Press ctrl+c to exit");
		ipcon.JoinThread();
    }
}
